### PR TITLE
Fix #17: Use WebDriver.executeAsyncScript to avoid timeouts.

### LINKED ISSUE #17: Potential timeouts while executing scripts.
When we execute a script using `WebDriver.executeScript` there is a potential for timeouts. 

A possible solution would be to use `WebDriver.executeAsyncScript`  and `WebDriver.Timeouts.setScriptTimeout` to avoid the timeouts as stated in #14.

https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/WebDriver.Timeouts.html
https://selenium.googlecode.com/svn/trunk/docs/api/java/org/openqa/selenium/JavascriptExecutor.html

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd.

**Comment by nicolasstucki** [comment]
To test this I created an application with an extremely long loop creating logs, it ran for a bit longer than half an hour with no timeouts.

**Comment by nicolasstucki** [comment]
Rebased.

**Comment by nicolasstucki** [comment]
Shouldn't have deleted this, itchy trigger finger.

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala [hunk 1]
```diff
@@ -11,6 +11,12 @@ import scala.collection.JavaConversions._
 import scala.collection.mutable.ArrayBuffer
 
 object Firefox extends SeleniumBrowser {
+
+  private final val popHijackedConsoleScript = {
+    "var callback = arguments[arguments.length - 1];" +
```
Comment (sjrd) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala:16]: Does this end up in the global scope? Can we avoid this with a wrapping like `(function() { ... }).call(this);`?

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala [hunk 2]
```diff
@@ -19,18 +19,27 @@ class SeleniumComJSRunner(browserProvider: SeleniumBrowser,
   protected def envName: String =
     "SeleniumComJSRunner on " + browserProvider.name
 
+  private final val sendScript = {
+    "var callback = arguments[arguments.length - 1];" +
```
Comment (sjrd) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala:23]: Same here, and in `receiveScript` below?

**Comment by sjrd** [comment]
That's all.

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala (hunk 1 above)
Comment (nicolasstucki) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala:16]: It does not end up in global scope, it is inserted in a closure that does `(function() { ... }).call(this);` and some other stuff to set up its own small environment.

Comment (nicolasstucki) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala:16]: It is not explicit in the documentation, but the documentation says to use it this way.
https://selenium.googlecode.com/svn/trunk/docs/api/java/org/openqa/selenium/JavascriptExecutor.html

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala (hunk 2 above)
Comment (nicolasstucki) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala:23]: Same

**Comment by nicolasstucki** [comment]
Rebased

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala (hunk 1 above)
Comment (sjrd) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala:16]: OK

**Comment by sjrd** [comment]
LGTM

**Pull request merged**